### PR TITLE
Docs (phase 1)

### PR DESCRIPTION
**Pull request by graebm**
- Use Sphinx to generate docs.
   It's very popular.

- Use Google comment style.
   Reads better than re-StructuredText. Shorter than Numpy style.

- Documented: io, mqtt

- Remaining work:
  - document: auth, http, exceptions
  - manual publish to Github pages
  - automatic publish to Github pages

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by graebm** [comment]
Here's what the generated docs look like:

Screen Shot 2020-05-07 at 11 01 10 PM

File: awscrt/mqtt.py [hunk 1]
```diff
@@ -21,10 +21,29 @@
...
+    """
+
     AT_LEAST_ONCE = 1
+    """QoS 1 - At least once delivery
+
+    This quality of service ensures that the message arrives at the receiver at least once.
+    """
+
     EXACTLY_ONCE = 2
+    """QoS 2 - Exactly once delivery
```
Comment (justinboswell) [inline awscrt/mqtt.py:41]: Should we note that this is not supported?

File: awscrt/mqtt.py [hunk 2]
```diff
@@ -21,10 +21,29 @@
 
 
 class QoS(IntEnum):
-    """Quality of Service"""
+    """Quality of Service enumeration"""
+
     AT_MOST_ONCE = 0
+    """QoS 0 - At most once delivery
```
Comment (justinboswell) [inline awscrt/mqtt.py:27]: Should we cite the spec sections like we do in native?

File: awscrt/mqtt.py [hunk 3]
```diff
@@ -67,6 +141,96 @@ def __init__(self, bootstrap, tls_ctx=None):
...
+            *   `session_present` (bool): True if resuming existing session. False if new session.
+                Note that the server has forgotten all previous subscriptions if this is False.
+                Subscriptions can be re-established via resubscribe_existing_topics().
+
+            *   `**kwargs` (dict): Forward-compatibility kwargs.
+
+        reconnect_min_timeout_secs (int): Minimum time to wait between reconnect attempts.
+            Wait starts at min and doubles with each attempt until max is reached.
+
+        reconnect_max_timeout_secs (int): Maximum time to wait between reconnect attempts.
```
Comment (justinboswell) [inline awscrt/mqtt.py:192]: Must be >= min?

File: awscrt/mqtt.py [hunk 4]
```diff
@@ -277,6 +373,12 @@ def on_connect(error_code, return_code, session_present):
         return future
 
     def reconnect(self):
+        """DEPRECATED.
```
Comment (justinboswell) [inline awscrt/mqtt.py:376]: Why don't we remove this?

File: awscrt/mqtt.py [hunk 5]
```diff
@@ -440,26 +634,33 @@ class WebsocketHandshakeTransformArgs(object):
...
-    They MUST call transform_args.set_done() when complete, passing an
-    exception if something went wrong. Failure to call set_done()
+    The function implementer may modify `transform_args.http_request` as desired.
+    They MUST call `transform_args.set_done()` when complete, passing an
+    exception if something went wrong. Failure to call `set_done()`
     will hang the application.
 
-    The implementor may do asynchronous work before calling transform_args.set_done(),
-    they are not required to call set_done() within the scope of the transform function.
+    The implementor may do asynchronous work before calling `transform_args.set_done()`,
```
Comment (justinboswell) [inline awscrt/mqtt.py:645]: Implementer. Implementor is the powerful antagonist in a programmer-centric comic book

File: awscrt/mqtt.py (hunk 1 above)
Comment (graebm) [reply awscrt/mqtt.py:41]: QoS 2 *is* supported in aws-c-mqtt, but not supported by the AWS IoT Core server. QoS 2 doesn't seem optional from my reading of the spec, but 😬

I'll throw in some cheeky language like "warning: some MQTT servers have been known to not support QoS 2"

File: awscrt/mqtt.py (hunk 2 above)
Comment (graebm) [reply awscrt/mqtt.py:27]: good idea

File: awscrt/mqtt.py (hunk 4 above)
Comment (graebm) [reply awscrt/mqtt.py:376]: We GA bro

File: awscrt/mqtt.py (hunk 5 above)
Comment (graebm) [reply awscrt/mqtt.py:645]: YOU MAKE IMPLEMENTOR ANGRY

**Pull request merged**